examples/tutorials/forced_alignment_for_multilingual_data_tutorial.py

```diff
@@ -0,0 +1,387 @@
+"""
+Forced alignment for multilingual data
+======================================
+
+**Author**: `Xiaohui Zhang <xiaohuizhang@meta.com>`__
+
+This tutorial shows how to compute forced alignments for speech data
+from multiple languages using ``torchaudio``\ ’s CTC forced alignment
+API described in `“CTC forced alignment
+tutorial” <https://pytorch.org/audio/stable/tutorials/forced_alignment_tutorial.html>`__
+and the multilingual Wav2vec2 model proposed in the paper `“Scaling
+Speech Technology to 1,000+
+Languages” <https://research.facebook.com/publications/scaling-speech-technology-to-1000-languages/>`__.
+The model was trained on 23K of audio data from 1100+ languages using
+the `“uroman vocabulary” <https://www.isi.edu/~ulf/uroman.html>`__
+as targets.
+
+"""
+
+import torch
+import torchaudio
+
+print(torch.__version__)
+print(torchaudio.__version__)
+
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print(device)
+
+
+try:
+    from torchaudio.functional import forced_align
+except ModuleNotFoundError:
+    print(
+        "Failed to import the forced alignment API. "
+        "Please install torchaudio nightly builds. "
+        "Please refer to https://pytorch.org/get-started/locally "
+        "for instructions to install a nightly build."
+    )
+    raise
+
+######################################################################
+# I. Preparation
+# --------------
+#
+# Here we import necessary packages, and define utility functions for
+# computing the frame-level alignments (using the API
+# ``functional.forced_align``), token-level and word-level alignments, and
+# also alignment visualization utilities.
+#
+
+# %matplotlib inline
+from dataclasses import dataclass
+
+import matplotlib
+import matplotlib.pyplot as plt
+import torchaudio.functional as F
+from IPython.display import Audio
+
+matplotlib.rcParams["figure.figsize"] = [16.0, 4.8]
+
+torch.random.manual_seed(0)
+
+sample_rate = 16000
+
+
+@dataclass
+class Frame:
+    # This is the index of each token in the transcript,
+    # i.e. the current frame aligns to the N-th character from the transcript.
+    token_index: int
+    time_index: int
+    score: float
+
+
+@dataclass
+class Segment:
+    label: str
+    start: int
+    end: int
+    score: float
+
+    def __repr__(self):
+        return f"{self.label}\t({self.score:4.2f}): [{self.start:5d}, {self.end:5d})"
+
+    @property
+    def length(self):
+        return self.end - self.start
+
+
+# compute frame alignments using torchaudio's forced alignment API
+def compute_alignments(transcript, dictionary, emission):
```

Comment:
Since the three helper functions, `compute_alignments`, `merge_repeats`, `merge_words` are already covered in the API tutorial, please merge these helper functions into one. We do not need to repeat calls to the same helper functions in this tutorial. This tutorial should forcus on "multi lingual" portion.

Reply:
sure